### PR TITLE
3234 table create manage views story

### LINKED ISSUE #3234: [table] create manage views story
### What package is this for?

- [x] React
- [ ] Angular

### Summary

Create a story group for table views management (create, read, update, delete). Part of #2820

**Additional context** Add any other context or screenshots about the feature
request here.

### Specific timeline issues / requests

Do you want this work within a specific time period? Is it related to an
upcoming release?

_NB: The core contributors will try to work with your timeline, but it's not
guaranteed. The earlier you make a request in advance of a desired delivery
date, the better!_

### Available extra resources

What resources do you have to assist this effort?

_Carbon add-ons for Watson IoT is a collaborative system. We encourage teams to
build components and submit them for integration._

### PR DESCRIPTION
**Pull request by bjornalm**
Closes #3234

**Summary**
Created story group for Manage views. This PR also solves #3243 

**Change List (commits, features, bugs, etc)**
- Create new story group
- Update/improve related stories 
- Create new mdx files, write docs and link from stories

**Acceptance Test (how to verify the PR)**
- Test that the table example at https://deploy-preview-3246--carbon-addons-iot-react.netlify.app?path=/story/1-watson-iot-table-user-view-management--with-user-view-management works as expected 
- Verify that the issue described in #3243 also is resolved
- Verify that the story links to to the proper mdx
- Verify that the mdx makes sense and that the links work
- Verify the component stories and docs 
  - https://deploy-preview-3246--carbon-addons-iot-react.netlify.app?path=/docs/1-watson-iot-table-user-view-management-tablemanageviewsmodal--with-callbacks-implemented
  - https://deploy-preview-3246--carbon-addons-iot-react.netlify.app?path=/docs/1-watson-iot-table-user-view-management-tablesaveviewmodal--playground
  - https://deploy-preview-3246--carbon-addons-iot-react.netlify.app?path=/docs/1-watson-iot-table-user-view-management-tableviewdropdown--playground

**Regression Test (how to make sure this PR doesn't break old functionality)**

- tests here

**Things to look for during review**

- [ ] Make sure all references to `iot` or `bx` class prefix is using the prefix variable
- [ ] (React) All major areas have a `data-testid` attribute. New test ids should have test written to ensure they are not changed or removed.
- [ ] UI should be checked in RTL mode to ensure the proper handling of layout and text.
- [ ] All strings should be translatable.
- [ ] The code should pass a11y scans (The storybook a11y knob should show no violations). New components should have a11y test written.
- [ ] Unit test should be written and should have a coverage of 90% or higher in all areas.
- [ ] All components should be passing visual regression test. For new styles or components either a visual regression test should be written for all permutations or the base image updated.
- [ ] Changes or new components should either write new or update existing documentation.
- [ ] PR should link and close out an existing issue

### CODE REVIEW

**Review by kevinsperrine — APPROVED** [review]
I've looked at this a couple of times. I don't see anything to address. Great work!

File: packages/react/src/components/Table/TableSaveViewModal/TableSaveViewModal.story.jsx [hunk 1]
```diff
@@ -34,18 +42,86 @@ export const DefaultWithKnobs = () => {
...
+      open={isOpen}
+      titleInputInvalid={titleInputInvalid}
+      titleInputInvalidText={titleInputInvalidText}
+      viewDescription={activeViewExample.description}
+    />
+  );
 };
 
-export const Prefilled = () => {
+WithDemoStates.storyName = 'With simple state management';
```
[the remark this answers is not in the document]
Comment (bjornalm) [reply packages/react/src/components/Table/TableSaveViewModal/TableSaveViewModal.story.jsx:116]: Good catch! I fixed it locally while solving some merge conflicts

File: packages/react/src/components/Table/TableManageViewsModal/TableManageViewsModalContent.mdx [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+| open                            | bool                            |                                                     | Determines if the modal is open or closed (i.e. visible or not to the user)                 |
+| overrides                       | shape                           | undefined                                           | Used to override the internal components and props for advanced customisation               |
+| pagination                      | shape                           | undefined                                           | pagination at the bottom of list, see SimplePaginationPropTypes                             |
+| testId                          | string                          | 'TableManageViewsModal'                             | Id that can be used for testing                                                             |
+| views                           | [ViewsPropType](#viewsproptype) |                                                     | The views to be displayed in the list, see ViewsPropType                                    |
```
Comment (davidicus) [inline packages/react/src/components/Table/TableManageViewsModal/TableManageViewsModalContent.mdx:120]: The link for ViewsPropType does not work.

Comment (bjornalm) [reply packages/react/src/components/Table/TableManageViewsModal/TableManageViewsModalContent.mdx]: Good catch, fixed

**Review by davidicus — APPROVED** [review]

**Pull request merged**